src/raftstore/store/msg.rs

```diff
@@ -18,11 +18,83 @@ use std::fmt;
 use kvproto::raft_serverpb::RaftMessage;
 use kvproto::raft_cmdpb::{RaftCmdRequest, RaftCmdResponse};
 use kvproto::metapb::RegionEpoch;
+
 use raft::SnapshotStatus;
 use util::escape;
 
-pub type Callback = Box<FnBox(RaftCmdResponse) + Send>;
-pub type BatchCallback = Box<FnBox(Vec<Option<RaftCmdResponse>>) + Send>;
+use super::engine::Snapshot;
+
+#[derive(Debug)]
+pub struct ReadArgs {
```

Comment:
How about `ReadResponse` instead of `ReadArgs`?

Reply:
+1